### ISSUE TITLE
[Premium] Unable to create vNET integration for existing Function App

### ISSUE THREAD

**Issue by andrewCluey** [Azure/Azure-Functions#1873]
**Describe the bug**
Using the suggested method for adding vNET integration to a Function App (here - https://github.com/Azure/azure-quickstart-templates/blob/master/101-function-premium-vnet-integration/azuredeploy.json), results in a succesful template deployment, but no vNET integration. 

**ARM resoruce**
```
       {
            "type": "Microsoft.Web/sites",
            "apiVersion": "2019-08-01",
            "name": "[variables('functionAppName')]",
            "location": "[parameters('location')]",
            "dependsOn": [
                "[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]",
                "[resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName'))]"

            ],
            "kind": "functionapp",
            "properties": {
                "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('hostingPlanName'))]",
                "siteConfig": {
                    "appSettings": [

                        {
                            "name": "AzureWebJobsStorage",
                            "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',variables('storageAccountName'), ';EndpointSuffix=', environment().suffixes.storage, ';AccountKey=',listkeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2019-06-01').keys[0].value,';')]"
                        },
                        {
                            "name": "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING",
                            "value": "[concat('DefaultEndpointsProtocol=https;AccountName=',variables('storageAccountName'), ';EndpointSuffix=', environment().suffixes.storage, ';AccountKey=',listkeys(resourceId('Microsoft.Storage/storageAccounts', variables('storageAccountName')), '2019-06-01').keys[0].value,';')]"
                        },
                        {
                            "name": "FUNCTIONS_EXTENSION_VERSION",
                            "value": "~3"
                        },
                        {
                            "name": "WEBSITE_VNET_ROUTE_ALL",
                             "value": "1"
                        },
                        {
                            "name": "FUNCTIONS_WORKER_RUNTIME",
                            "value": "[variables('functionWorkerRuntime')]"
                        },
                        {
                            "name": "WEBSITE_NODE_DEFAULT_VERSION",
                            "value": "~12"
                        }
                    ]
                },
                "resources": [
                    {
                        "type": "networkConfig",
                        "apiVersion": "2019-08-01",
                        "name": "virtualNetwork",
                        "location": "[parameters('location')]",
                        "dependsOn": [
                            "[resourceId('Microsoft.Web/sites/', variables('functionAppName'))]"
                            ],
                        "properties": {
                            "subnetResourceId": "[parameters('subnetId')]",
                            "isSwift": true
                        }
                    }
                ]
            }
```

**Expected behavior**
vNET integration to be configured for the Function.

**Screenshots**

![image](https://user-images.githubusercontent.com/13432888/113736577-21040e80-96f5-11eb-8dd1-63ad6fa066bb.png)

**Comment by lecamarade** [comment]
Same here. I get: "Regional VNET Integration network cannot be modified with this API call. Please use call /networkConfig/VirtualNetwork."

Have you found a work around?

The sample you mentioned is no longer available, btw. Wonder if it's related to this bug.


I was hoping it was ARM schema related. I get apiVersion 2018-11-01 when I export it in the UI.
This is what it looks like:

{
	"type": "Microsoft.Web/sites/config",
	"apiVersion": "2018-11-01",
...
	"properties": {
		"vnetName": "myVnetName",
...   
	}
},

  {
	"type": "Microsoft.Web/sites/virtualNetworkConnections",
	"apiVersion": "2018-11-01",
	"name": "[concat(parameters('siteName'), '/myVnetName')]",
	"location": "West US",
	"properties": {
		"vnetResourceId": "[concat(parameters('vnet_external_id'), '/subnets/mysubnet')]",
		"isSwift": true
	}
}

**Comment by v-anvari** [comment]
Hi @andrewCluey , Apologies for the delayed response, does this example help in your scenario https://github.com/ScottHolden/ARMExamples/blob/master/AppService-VnetNew-Storage/deploy.json

@lecamarade , Can we have the repro steps that you are trying in your project

**Comment by lecamarade** [comment]
I had to use the format below to make it work. Vnet was added succesfully.
```
{
   "type": "Microsoft.Web/sites/networkConfig",
   "name": "[concat(parameters('webAppName'),'/VirtualNetwork')]",
   "apiVersion": "2016-08-01",
   "properties": 
   {
       "subnetResourceId": "[parameters('subnetResourceId')]"
   }
}

```

**Comment by timomta** [comment]
I had this same problem. I did get the [specific sample](https://github.com/ScottHolden/ARMExamples/blob/master/AppService-VnetNew-Storage/deploy.json) pointed to above to work. This really is a confusing issue because if you set this up from the portal, the exported template will fail and there isn't any doc (that I found) that shows you how to do vnet integration via ARM.
Now that I know the ScottHolden sample works, I have a baseline I hope to integrate with my templates.
